### PR TITLE
Let 352 2 fe project creation form

### PR DESCRIPTION
**Pull request by barbara-chaves**
Project creation form - All pages + leave modal + pending page

### Acceptance criteria
There will be a way for a signed up user with role “project developer” to create a project

The user is able to enter the [project’s mandatory and optional fields](https://docs.google.com/spreadsheets/d/1qtP-qkm8iS6OlluiJwRvQoi6rCns9iKhRYoNzc9qkGg/edit#gid=1884416048) (see “project” fields in the “forms” sheet)

There will be a way to validate the information entered in the fields

Errors are displayed if the information is incorrect 

There will be a way for the user to know that the content should be written in the language of the account

There will be a way confirm or dismiss the creation of the project

Note: Save/cancel

The layout must comply with the designs defined [HeCo - Work in progress](https://www.figma.com/proto/KC49uMR9t1YARmFR0Sr6aN/HeCo---Work-in-progress?node-id=1702%3A24167&viewport=396%2C48%2C0.06&scaling=scale-down&starting-point-node-id=1702%3A24167&show-proto-sidebar=1)

## Tracking

[LET-352](https://vizzuality.atlassian.net/browse/LET-352)

### CODE REVIEW

File: frontend/pages/projects/new.tsx [hunk 1]
```diff
@@ -59,28 +80,32 @@ const Project: PageComponent<ProjectProps, FormPageLayoutProps> = () => {
...
-      //     push('/project-developers/pending');
-      //   },
-      // }),
-      console.log(data),
-    []
+    (data: ProjectCreationPayload) =>
+      createProject.mutate(data, {
+        onError: (error) => {
+          const { errorPages, fieldErrors } = getServiceErrors<ProjectForm>(error, formPageInputs);
+          console.log(error, errorPages, fieldErrors);
```
Comment (clementprdhomme) [inline frontend/pages/projects/new.tsx:87]:
```suggestion
```

File: frontend/pages/projects/new.tsx [hunk 2]
```diff
@@ -59,28 +80,32 @@ const Project: PageComponent<ProjectProps, FormPageLayoutProps> = () => {
...
+          fieldErrors.forEach((field) => setError(field, { message: '' }));
```
Comment (clementprdhomme) [inline frontend/pages/projects/new.tsx:88]: What's the purpose of this line? Isn't this displaying empty error messages?

File: frontend/schemas/project.ts [hunk 3]
```diff
@@ -4,6 +4,14 @@ import { object, string, array, number, mixed, boolean } from 'yup';
 
 export default (page: number) => {
   const { formatMessage } = useIntl();
+  const maxTextLength = formatMessage({
+    defaultMessage: 'It must have a maximum of 600 characters',
+    id: 'frm1UB',
+  });
+  const booleanField = formatMessage({
+    defaultMessage: 'Yo need to select an option',
```
Comment (clementprdhomme) [inline frontend/schemas/project.ts:12]:
```suggestion
    defaultMessage: 'You need to select an option',
```

File: frontend/schemas/project.ts [hunk 4]
```diff
@@ -14,6 +22,7 @@ export default (page: number) => {
       id: '+m/R6q',
     }),
     project_gallery: {
+      min_lenth: formatMessage({ defaultMessage: 'Upload at least one picture', id: 'BG1UW6' }),
```
Comment (clementprdhomme) [inline frontend/schemas/project.ts:25]: This was removed in the previous PR.
```suggestion
```

**Review by clementprdhomme — COMMENTED** [review]
In addition to my comments below:
- Page 3 (Impact) is not validated: the user can click continue without selecting anything
- Page 4, “From which investor or funder?” is optional but it's not indicated in the UI with “(optional)” in the label. Shouldn't we add it?

**Comment by clementprdhomme** [comment]
@barbara-chaves If not part of this PR, could you create another one with these points:
- There will be a way for the user to know that the content should be written in the language of the account
- There will be a way confirm or dismiss the creation of the project

File: frontend/containers/project-form-pages/pages/project-description.tsx [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+          <div className="mt-4 mb-8">
+            <fieldset>
+              <legend className="inline font-sans font-semibold text-sm text-gray-800 mb-4.5">
+                <span className="mr-2.5">
+                  <FormattedMessage
+                    defaultMessage="Which of these topics/sector categories better describe your project?"
+                    id="i2AgQl"
+                  />
+                </span>
+                <FieldInfo infoText="MISSING" />
```
[the remark this answers is not in the document]
Comment (barbara-chaves) [reply frontend/containers/project-form-pages/pages/project-description.tsx:128]: Yes, I forgot that it was already in the categories enum

File: frontend/pages/projects/new.tsx (hunk 2 above)
Comment (barbara-chaves) [reply frontend/pages/projects/new.tsx:87]: Yes, it is not finished. I will remove it for now

**Pull request merged**